contracts/voting_escrow/src/state.rs

```diff
@@ -0,0 +1,61 @@
+use cosmwasm_std::{Addr, CanonicalAddr, Decimal, Uint128};
+use cw_storage_plus::{Item, Map, U64Key};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// ## Description
+/// This structure describes the main control config of voting escrow contract.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Config {
+    /// contract address use for settings control
+    pub owner: CanonicalAddr,
+    /// ANC token address
+    pub anchor_token: CanonicalAddr,
+}
+
+/// ## Description
+/// This structure describes the point in checkpoints history.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Point {
+    /// voting power
+    pub power: Uint128,
+    /// equals to the point period
+    pub start: u64,
+    /// the period when the lock should expire
+    pub end: u64,
+    /// voting power decay per period at the current period
+    pub slope: Decimal,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Lock {
+    /// the total ANC tokens were deposited
+    pub amount: Uint128,
+    /// the period when lock was created
+    pub start: u64,
+    /// the period when the lock should expire
+    pub end: u64,
+    /// the last period when the lock's time was increased
+    pub last_extend_lock_period: u64,
+}
+
+/// ## Description
+/// Stores config at the given key
+pub const CONFIG: Item<Config> = Item::new("config");
+
+/// ## Description
+/// Stores all user locks
+pub const LOCKED: Map<Addr, Lock> = Map::new("locked");
+
+/// ## Description
+/// Stores checkpoint history per composed key (addr, period).
+/// Total voting power checkpoints are stored by (contract_addr, period) key.
+pub const HISTORY: Map<(Addr, U64Key), Point> = Map::new("history");
```

Comment:
Yeah, I was wondering whether I should change it... I don't think the storage to use is standardized though... I noticed we use `cw_storage_plus` in a very recent set of contracts https://github.com/Anchor-Protocol/crossanchor-monorepo/tree/main/deployment/terra/contracts

Reply:
we should use `cw_storage_plus`, we've been trying to move away from `cosmwasm_storage` but still use it in some places because we wrote those contracts before `cw_storage_plus` existed

Reply:
Oh I didn't know it. I used `cosmwasm_storage` in `gauge controller contract`, does it mean that I need to change it to `cw_storage_plus`?

Reply:
hey @cdocu11 `cw_storage_plus` is the current standard; lets change if its easy

Reply:
Yeah, I'll do this.